crates/core/src/client/mod.rs

```diff
@@ -372,13 +372,23 @@ impl<T: JsonRpcTransport + Send + Sync> KakarotEthApi<T> for KakarotClient<JsonR
     }
 
     /// Returns the nonce for a given ethereum address
+    /// if ethereum -> stark mapping doesn't exist in the starknet provider, we translate
+    /// ContractNotFound errors into zeros
     async fn nonce(&self, ethereum_address: Address, block_id: BlockId) -> Result<U256, EthApiError<T::Error>> {
         let starknet_block_id = EthBlockId::new(block_id).to_starknet_block_id::<T>()?;
         let starknet_address = self.compute_starknet_address(ethereum_address, &starknet_block_id).await?;
 
-        let nonce: Felt252Wrapper = self.starknet_provider.get_nonce(starknet_block_id, starknet_address).await?.into();
-
-        Ok(nonce.into())
+        self.starknet_provider
+            .get_nonce(starknet_block_id, starknet_address)
+            .await
+            .map(|nonce| {
+                let nonce: Felt252Wrapper = nonce.into();
+                nonce.into()
+            })
+            .or_else(|err| match err {
+                ProviderError::StarknetError(StarknetError::ContractNotFound) => Ok(U256::from(0)),
```

Comment:
would it be useful to have dev level logging that prints 'address not found' specifically? i think it probably would be. 

are we doing logging like this anywhere?

Reply:
Ahh good question, I guess not!
We have not setup a dev logging system yet, apart from the tracing subscriber

Reply:
when trying to deploy kakaswap this was one of the error cases i was first hitting when the rpc would 'throw' on trying to get a nonce. having a clear signal to the brave dev attempting to use it for tooling that 'hey, this address doesnt exist, did you deploy?

Reply:
approve that non existing EOAccount is n°1 issue for anyone trying to use kakarot-rpc. In the python sandbox, I added a Starknet account owned by the RPC to deploy if required EOAccount.

We don't need to always deploy (like returning 0 is just the good thing to do for nonce), but trying to send a tx, we should, at least in the beginning.

From a UX point of viw, we should not have EOAccount with balance > 0 but not deployed.

Reply:
Also agree on the need for a logger